coco_architecture.md

```diff
@@ -0,0 +1,147 @@
+- **Data integrity:** Unauthorized entities cannot add, remove, or alter data while it is in use within the TEE
+- **Code integrity:** Unauthorized entities cannot add, remove, or alter code executing in the TEE
+
+Ref: https://confidentialcomputing.io/wp-content/uploads/sites/85/2021/03/confidentialcomputing_outreach_whitepaper-8-5x11-1.pdf
+
+# Confidential Computing (CC) Building blocks 
+
+## Overview of the key components of a confidential computing solution
+
+The following diagrams shows how the different building blocks for CC come together: 
+
+![CC_all_blocks](https://github.com/confidential-containers/documentation/blob/main/images/CC_all_blocks.jpg)
+
+Let’s start by describing the lower level blocks of the CC solution (**colored blue**): 
+
+- **Infra layer** - be it on prem (bare metal, VMware etc..) or public cloud(AWS, Azure, GCP etc…)
+  - **Hardware with CC support** - this includes 2 areas of technologies
+  - **Process based TEEs** - In this model, a process that needs to run securely is divided into two components: trusted and untrusted. 
+    The trusted component resides in encrypted memory and handles confidential computing, while the untrusted component interfaces with 
+    the operating system and propagates I/O from encrypted memory to the rest of the system. Intel SGX is an example of a process based TEE
+  - **VM based TEEs**  - In this model, memory is encrypted along a traditional VM boundary running on top of a VMM. AMD SEV-SNP, Intel TDX, 
+    IBM Secure Execution and Protected Execution Functionality (PEF) are examples of VM TEEs.
+- **Hypervisor** - this includes hypervisors such as QEMU/KVM, cloud hypervisor, public cloud provider hypervisors etc...  
+
+The **CC services block** contains a number of services (**colored pink**) which are required for creating a holistic CC platform which the customer 
+can then use to build their solution. This includes the following services (we expect to see more services later on):  
+
+- **Attestation service** -  The primary purpose of the attestation service is to validate the evidence provided by the hardware TEE. 
+  This is the verifier as defined in the RATS architecture.
+- **Key Broker Service (KBS)** -  The KBS is the relying party as defined in the RATS architecture.  Following are its primary function:
+  - Receives evidence from the attester (confidential VM or container) via challenge-response protocol
+  - Relay the evidence to the Attestation Service for verification
+  - Apply appraisal policy for the returned Attestation Results to assess the trustworthiness of the attester
+  - Interact with Key Management Service to retrieve the keys and send it to the attester
+- **Key management service** - A service for securely storing, managing and backing up of cryptographic keys used by applications and users.
+- **Image build service** - A service that is used to build confidential container or VM images for end users
```

Comment:
Building VM images is another can of worms. Not sure we want to reference that here.

Reply:
Ah think i added my comment in this area to updated code. Hopefully people can correlate it to here.